packages/compose/src/hooks/use-copy-on-click/index.js

```diff
@@ -30,45 +25,67 @@ export default function useCopyOnClick( ref, text, timeout = 4000 ) {
 		alternative: 'wp.compose.useCopyToClipboard',
 	} );
 
-	/** @type {import('react').MutableRefObject<Clipboard | undefined>} */
-	const clipboard = useRef();
 	const [ hasCopied, setHasCopied ] = useState( false );
 
 	useEffect( () => {
 		/** @type {number | undefined} */
 		let timeoutId;
+		/** @type Array<Element>) */
```

Comment:
Out of curiosity, does the `compose` package not support TypeScript directly? I haven't seen this syntax before :) Is it jsdoc?

Reply:
https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html

Reply:
As Zebulan pointed out, these are type hints for the TypeScript server written with the JSDoc syntax — I decided to adopt this syntax to match other files in the `@wordpress/compose` package — although we should definitely consider migrating the whole package to TypeScript in the future.